src/toPlaylists.js

```diff
@@ -16,18 +11,14 @@ export const generateSegments = (segmentInfo, attributes) => {
     );
   }
 
-  // TODO
   if (segmentInfo.base) {
-    throw new Error(errors.UNSUPPORTED_SEGMENTATION_TYPE);
-
-    // return segmentsFromBase(attributes);
+    return segmentsFromBase(shallowMerge(segmentInfo.base, attributes));
   }
 
-  // TODO
   if (segmentInfo.list) {
-    throw new Error(errors.UNSUPPORTED_SEGMENTATION_TYPE);
-
-    // return segmentsFromList(attributes);
+    return segmentsFromList(
+      shallowMerge(segmentInfo.list, attributes), segmentInfo.timeline
```

Comment:
Can anything in `attributes` overlap with `segmentInfo.list` (and `segmentInfo.base` for above)? If so we probably want to swap the order of the arguments so any overlapping values take from `segmentInfo`. If not then no need

Reply:
Which makes me realize, I should double check for segment template

Reply:
In either case it seems safer to swap the order.

Reply:
It doesn't look like there will be any overlap. Depending on how `Initialization` tag is handled, will have to keep that in mind for `SegmentTemplate` since it can have both the tag and attribute. I don't think we will encounter a manifest with SegmentTemplate that has both the attribute and tag, but if we do, we will want to make a choice on which takes precedence.

Reply:
We shouldn't encounter that scenario, but it's a possibility.  I didn't see anything in the spec with guidance where `@initialization` and `<Initialization>` relate to the same `<SegmentTemplate>`.  So it seems it's up to our discretion.  I made an assumption that if both are provided that the `<Initialization>` will take precedence (and added a comment noting as much).  Given the tree structure I think that makes sense.  

The other options are to just error if this is the case (less permissive, and it is probably easier to move from less permissive -> more permissive) or `@initialization` taking precedence.

Reply:
Makes sense to me 👍